### ISSUE TITLE
Ubuntu 16.04 RStudio Freeze when typing "Enter", "delete", or "Ctrl+delete"

### ISSUE THREAD

**Issue by YoannPa** [rstudio/rstudio#4357]
### System details

    RStudio Edition : Desktop
    Version : 1.1.463
    OS Version : (Linux) Ubuntu 16.04
    R Version : 3.5.2 (2018-12-20) -- "Eggshell Igloo"

Hi,
I have a huge issue with RStudio: I was coding using 2 packages in 2 different sessions (TCGAbiolinks and RnBeads). Then suddenly for no reason typing "Enter" or "delete" or "Ctrl+delete" on my keyboard made freeze several Rstudio.
I decided to restart R from the interface first: it didn't work.
Then I closed all my Rstudio session and interfaces, to reopen one and see: it didn't work.
Then I decided to restart my computer: it didn't work.
Then I decided to remove the entire folder ~.rstudio-desktop/ and to uninstall RStudio with `sudo apt-get remove rstudio`. I reinstalled Rstudio using the deb file `rstudio-xenial-1.1.463-amd64.deb`. It didn't work. 

This is the very first time I experience such issue in 5 years. I haven't been able to find any related topic on the web about the issue I face.

The Freezing of the Rstudio interface is associated to CPU use of 1 core (100%) by a PID rstudio while it freeze.

1. Here is the output of sessionInfo()
```
R version 3.5.2 (2018-12-20)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.18.so

locale:
 [1] LC_CTYPE=fr_FR.UTF-8       LC_NUMERIC=C               LC_TIME=fr_FR.UTF-8        LC_COLLATE=fr_FR.UTF-8    
 [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=fr_FR.UTF-8    LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
  [1] colorspace_1.4-0            selectr_0.4-1               rjson_0.2.20                hwriter_1.3.2              
  [5] circlize_0.4.5              XVector_0.22.0              GenomicRanges_1.34.0        GlobalOptions_0.1.0        
  [9] rstudioapi_0.9.0            ggpubr_0.2                  matlab_1.0.2                ggrepel_0.8.0              
 [13] bit64_0.9-7                 AnnotationDbi_1.44.0        xml2_1.2.0                  codetools_0.2-16           
 [17] splines_3.5.2               R.methodsS3_1.7.1           doParallel_1.0.14           DESeq_1.34.1               
 [21] geneplotter_1.60.0          knitr_1.21                  jsonlite_1.6                Rsamtools_1.34.1           
 [25] km.ci_0.5-2                 broom_0.5.1                 annotate_1.60.0             cluster_2.0.7-1            
 [29] R.oo_1.22.0                 readr_1.3.1                 compiler_3.5.2              httr_1.4.0                 
 [33] backports_1.1.3             assertthat_0.2.0            Matrix_1.2-15               lazyeval_0.2.1             
 [37] limma_3.38.3                prettyunits_1.0.2           tools_3.5.2                 bindrcpp_0.2.2             
 [41] gtable_0.2.0                glue_1.3.0                  GenomeInfoDbData_1.2.0      dplyr_0.7.8                
 [45] ggthemes_4.0.1              ShortRead_1.40.0            Rcpp_1.0.0                  Biobase_2.42.0             
 [49] Biostrings_2.50.2           nlme_3.1-137                rtracklayer_1.42.1          iterators_1.0.10           
 [53] xfun_0.4                    stringr_1.4.0               rvest_0.3.2                 XML_3.98-1.17              
 [57] edgeR_3.24.3                zoo_1.8-4                   zlibbioc_1.28.0             scales_1.0.0               
 [61] aroma.light_3.12.0          hms_0.4.2                   parallel_3.5.2              SummarizedExperiment_1.12.0
 [65] RColorBrewer_1.1-2          ComplexHeatmap_1.20.0       yaml_2.2.0                  memoise_1.1.0              
 [69] gridExtra_2.3               KMsurv_0.1-5                ggplot2_3.1.0               downloader_0.4             
 [73] biomaRt_2.38.0              latticeExtra_0.6-28         stringi_1.2.4               RSQLite_2.1.1              
 [77] genefilter_1.64.0           S4Vectors_0.20.1            foreach_1.4.4               GenomicFeatures_1.34.3     
 [81] BiocGenerics_0.28.0         BiocParallel_1.16.6         shape_1.4.4                 GenomeInfoDb_1.18.1        
 [85] rlang_0.3.1                 pkgconfig_2.0.2             matrixStats_0.54.0          bitops_1.0-6               
 [89] lattice_0.20-38             purrr_0.3.0                 bindr_0.1.1                 cmprsk_2.2-7               
 [93] GenomicAlignments_1.18.1    bit_1.1-14                  tidyselect_0.2.5            plyr_1.8.4                 
 [97] magrittr_1.5                R6_2.3.0                    IRanges_2.16.0              generics_0.0.2             
[101] TCGAbiolinks_2.10.3         DelayedArray_0.8.0          DBI_1.0.0                   mgcv_1.8-27                
[105] pillar_1.3.1                survival_2.43-3             RCurl_1.95-4.11             tibble_2.0.1               
[109] EDASeq_2.16.3               crayon_1.3.4                survMisc_0.5.5              GetoptLong_0.1.7           
[113] progress_1.2.0              locfit_1.5-9.1              grid_3.5.2                  sva_3.30.1                 
[117] data.table_1.12.0           blob_1.1.1                  ConsensusClusterPlus_1.46.0 digest_0.6.18              
[121] xtable_1.8-3                tidyr_0.8.2                 R.utils_2.7.0               stats4_3.5.2               
[125] munsell_0.5.0               survminer_0.4.3
```

 
2. The R code in question
```
##IMPORTS
source("/home/yoann/get_TCGAproj_biotab.R")

##PARAMETERS
Clinical.dir<-"/home/yoann/TCGA"
Projects_list<-TCGAbiolinks:::getGDCprojects()$project_id
Biotab.dir<-"/home/yoann/TCGA/Biotabs"

##ANALYSIS
#Get projects biotabs
res<-get_TCGAproj_biotab(project.id = Projects_list,
                          clin.data.dir = Clinical.dir)

#Save Biotabs in separated rds files
if(dir.exists(Biotab.dir) == F){
  dir.create(Biotab.dir)
}

(seq_along(1:length(res$Biotabs)),function(i) {saveRDS(res$Biotabs[[i]],
                                                             file = paste0(Clinical.dir,
                                                                           "/Biotabs/",names(res$Biotabs[i]), ".RDS"))})
```

Here is the content of **/home/yoann/get_TCGAproj_biotab.R** :
```
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(simsalapar)


get_TCGAproj_biotab<-function(project.id,clin.data.dir){
  #Keep only TCGA projects
  processable.proj<-project.id[grepl("^TCGA-*",project.id)]
  
  #Display projects that are not from TCGA
  unprocessable.proj<-project.id[!(project.id %in% processable.proj)]
  cat(paste(c("WARNING! Cannot process the following projects:",
              unprocessable.proj,"\n")))
  #Change working dir
  main_wd<-getwd()
  setwd(clin.data.dir)
  #Set lists
  list.biotab<-list()
  list.warn.query<-list()
  list.warn.DL<-list()
  list.warn.prep<-list()
  #Download TCGA projects clinical data and create biotabs
  for (proj in processable.proj) {
    cat(paste0(proj,"\n"))
    #Create GDC query
    list.warn.query[[proj]] <- tryCatch.W.E(query <- GDCquery(project = proj,
                                                    data.category = "Clinical",
                                                    data.type = "Clinical data",
                                                    legacy = TRUE,
                                                    file.type = "txt"))$warning
    #Download Clin. data
    list.warn.DL[[proj]] <-tryCatch.W.E(GDCdownload(query))$warning
    #Get clinical biotab 
    list.warn.prep[[proj]] <- tryCatch.W.E(Prep.query <-
                                             GDCprepare(query))$warning
    #Store clin.biotab
    list.biotab[[proj]]<-Prep.query
  }
  #Move back to original Working dir 
  setwd(main_wd)
  #Create result with biotabs and warnings
  res<-list("Biotabs" = list.biotab,
            "unprocessed.projects" = unprocessable.proj,
            "warnings.query" = list.warn.query, "warnings.DL" = list.warn.DL,
            "warnings.preps" = list.warn.prep)
  return(res)
}
```

3. The diagnostics report:

[1] "1.1.463"

$R
[1] "/usr/bin/R"

$pdflatex
[1] "/usr/bin/pdflatex"

$bibtex
[1] "/usr/bin/bibtex"

$gcc
[1] "/usr/bin/gcc"

$git
[1] "/usr/bin/git"

$svn
[1] ""

R version 3.5.2 (2018-12-20)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.18.so

locale:
 [1] LC_CTYPE=fr_FR.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=fr_FR.UTF-8        LC_COLLATE=fr_FR.UTF-8    
 [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=fr_FR.UTF-8   
 [7] LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.5.2 tools_3.5.2   

SysInfo:
                                       sysname 
                                       "Linux" 
                                       release 
                           "4.4.0-142-generic" 
                                       version 
"#168-Ubuntu SMP Wed Jan 16 21:00:45 UTC 2019" 
                                      nodename 
                               "yoann-desktop" 
                                       machine 
                                      "x86_64" 
                                         login 
                                     "unknown" 
                                          user 
                                       "yoann" 
                                effective_user 
                                       "yoann" 

R Version:
               _                           
platform       x86_64-pc-linux-gnu         
arch           x86_64                      
os             linux-gnu                   
system         x86_64, linux-gnu           
status                                     
major          3                           
minor          5.2                         
year           2018                        
month          12                          
day            20                          
svn rev        75870                       
language       R                           
version.string R version 3.5.2 (2018-12-20)
nickname       Eggshell Igloo              
$`_`
[1] "/usr/bin/rstudio"

$CLICOLOR_FORCE
[1] "1"

$CLUTTER_IM_MODULE
[1] "xim"

$COMPIZ_BIN_PATH
[1] "/usr/bin/"

$COMPIZ_CONFIG_PROFILE
[1] "ubuntu"

$DBUS_SESSION_BUS_ADDRESS
[1] "unix:abstract=/tmp/dbus-XiMelWFVri"

$DEFAULTS_PATH
[1] "/usr/share/gconf/ubuntu.default.path"

$DESKTOP_SESSION
[1] "ubuntu"

$DISPLAY
[1] ":0"

$EDITOR
[1] "vi"

$GDM_LANG
[1] "fr_FR"

$GDMSESSION
[1] "ubuntu"

$GIT_ASKPASS
[1] "rpostback-askpass"

$GNOME_DESKTOP_SESSION_ID
[1] "this-is-deprecated"

$GNOME_KEYRING_CONTROL
[1] ""

$GNOME_KEYRING_PID
[1] ""

$GPG_AGENT_INFO
[1] "/home/yoann/.gnupg/S.gpg-agent:0:1"

$GTK2_MODULES
[1] "overlay-scrollbar"

$GTK_IM_MODULE
[1] "ibus"

$GTK_MODULES
[1] "gail:atk-bridge:unity-gtk-module"

$HOME
[1] "/home/yoann"

$IM_CONFIG_PHASE
[1] "1"

$INSTANCE
[1] ""

$JOB
[1] "unity-settings-daemon"

$LANG
[1] "fr_FR.UTF-8"

$LANGUAGE
[1] "fr_FR"

$LD_LIBRARY_PATH
[1] "/usr/lib/R/lib::/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/default-java/jre/lib/amd64/server"

$LESSCLOSE
[1] "/usr/bin/lesspipe %s %s"

$LESSOPEN
[1] "| /usr/bin/lesspipe %s"

$LN_S
[1] "ln -s"

$LOGNAME
[1] "yoann"

$LS_COLORS
[1] "rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:"

$MAKE
[1] "make"

$MANDATORY_PATH
[1] "/usr/share/gconf/ubuntu.mandatory.path"

$PAGER
[1] "/usr/bin/pager"

$PATH
[1] "/home/yoann/bin:/home/yoann/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"

$PWD
[1] "/home/yoann"

$QT4_IM_MODULE
[1] "xim"

$QT_ACCESSIBILITY
[1] "1"

$QT_IM_MODULE
[1] "ibus"

$QT_LINUX_ACCESSIBILITY_ALWAYS_ON
[1] "1"

$QT_QPA_PLATFORMTHEME
[1] "appmenu-qt5"

$R_BROWSER
[1] "xdg-open"

$R_BZIPCMD
[1] "/bin/bzip2"

$R_DOC_DIR
[1] "/usr/share/R/doc"

$R_GZIPCMD
[1] "/bin/gzip -n"

$R_HOME
[1] "/usr/lib/R"

$R_INCLUDE_DIR
[1] "/usr/share/R/include"

$R_LIBS_SITE
[1] "/usr/local/lib/R/site-library:/usr/lib/R/site-library:/usr/lib/R/library"

$R_LIBS_USER
[1] "~/R/x86_64-pc-linux-gnu-library/3.5"

$RMARKDOWN_MATHJAX_PATH
[1] "/usr/lib/rstudio/resources/mathjax-26"

$R_PAPERSIZE
[1] "letter"

$R_PAPERSIZE_USER
[1] "a4"

$R_PDFVIEWER
[1] "/usr/bin/xdg-open"

$R_PLATFORM
[1] "x86_64-pc-linux-gnu"

$R_PRINTCMD
[1] "/usr/bin/lpr"

$R_RD4PDF
[1] "times,inconsolata,hyper"

$R_SESSION_TMPDIR
[1] "/tmp/RtmpUPjl1I"

$R_SHARE_DIR
[1] "/usr/share/R/share"

$RS_LOCAL_PEER
[1] "/tmp/24125-rsession"

$RS_PPM_FD_READ
[1] "20"

$RS_PPM_FD_WRITE
[1] "21"

$RS_RPOSTBACK_PATH
[1] "/usr/lib/rstudio/bin/rpostback"

$RS_SHARED_SECRET
[1] "16816927778469308861804289383"

$RSTUDIO
[1] "1"

$RSTUDIO_CONSOLE_COLOR
[1] "256"

$RSTUDIO_CONSOLE_WIDTH
[1] "80"

$RSTUDIO_PANDOC
[1] "/usr/lib/rstudio/bin/pandoc"

$RSTUDIO_SESSION_PORT
[1] "24125"

$RSTUDIO_USER_IDENTITY
[1] "yoann"

$RSTUDIO_WINUTILS
[1] "bin/winutils"

$R_SYSTEM_ABI
[1] "linux,gcc,gxx,gfortran,?"

$R_TEXI2DVICMD
[1] "/usr/bin/texi2dvi"

$R_UNZIPCMD
[1] "/usr/bin/unzip"

$R_ZIPCMD
[1] "/usr/bin/zip"

$SED
[1] "/bin/sed"

$SESSION
[1] "ubuntu"

$SESSIONTYPE
[1] "gnome-session"

$SHELL
[1] "/bin/bash"

$SHLVL
[1] "1"

$SSH_AUTH_SOCK
[1] "/run/user/1000/keyring/ssh"

$TAR
[1] "/bin/tar"

$TERM
[1] "xterm-256color"

$UNITY_DEFAULT_PROFILE
[1] "unity"

$UNITY_HAS_3D_SUPPORT
[1] "true"

$UPSTART_EVENTS
[1] "xsession started"

$UPSTART_INSTANCE
[1] ""

$UPSTART_JOB
[1] "unity7"

$UPSTART_SESSION
[1] "unix:abstract=/com/ubuntu/upstart-session/1000/1627"

$USER
[1] "yoann"

$VTE_VERSION
[1] "4205"

$WINDOWID
[1] "77594634"

$XAUTHORITY
[1] "/home/yoann/.Xauthority"

$XDG_CONFIG_DIRS
[1] "/etc/xdg/xdg-ubuntu:/usr/share/upstart/xdg:/etc/xdg"

$XDG_CURRENT_DESKTOP
[1] "Unity"

$XDG_DATA_DIRS
[1] "/usr/share/ubuntu:/usr/share/gnome:/usr/local/share:/usr/share:/var/lib/snapd/desktop:/var/lib/snapd/desktop"

$XDG_GREETER_DATA_DIR
[1] "/var/lib/lightdm-data/yoann"

$XDG_MENU_PREFIX
[1] "gnome-"

$XDG_RUNTIME_DIR
[1] "/run/user/1000"

$XDG_SEAT
[1] "seat0"

$XDG_SEAT_PATH
[1] "/org/freedesktop/DisplayManager/Seat0"

$XDG_SESSION_DESKTOP
[1] "ubuntu"

$XDG_SESSION_ID
[1] "c1"

$XDG_SESSION_PATH
[1] "/org/freedesktop/DisplayManager/Session0"

$XDG_SESSION_TYPE
[1] "x11"

$XDG_VTNR
[1] "7"

$XMODIFIERS
[1] "@im=ibus"

 [1] ".GlobalEnv"        "tools:rstudio"     "package:stats"    
 [4] "package:graphics"  "package:grDevices" "package:utils"    
 [7] "package:datasets"  "package:methods"   "Autoloads"        
[10] "package:base"     
Log file: rdesktop.log
--------------------------------------------------

(Empty)

Log file: rsession-yoann.log
--------------------------------------------------

(Empty)

**Comment by pascaloettli** [comment]
Do you reproduce in the preview release?

**Comment by YoannPa** [comment]
@pascaloettli No I did not.
I am going to close this issue since I achieve to solve the problem.
Unfortunately I have still no what was the origin of the problem.
All I can say is that since I posted this issue, I deleted the .Rhistory file. So far the problem seems to be solved.
I will wait one more week because I will have to work with these 2 packages for some more time (TCGAbiolinks and RnBeads) if after 1 week the issue does not show up again, I will close it. Otherwise I will save the Rhistory before deleting it, and will post it so that everybody can look for the origin of the problem.
